sklearn/utils/tests/test_stats.py

```diff
@@ -48,23 +66,35 @@ def test_weighted_percentile_zero_weight_zero_percentile():
     assert approx(score) == 4
 
 
-def test_weighted_median_equal_weights():
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype_name", yield_namespace_device_dtype_combinations()
+)
+def test_weighted_median_equal_weights(array_namespace, device, dtype_name):
+    xp = _array_api_for_tests(array_namespace, device)
     # Checks weighted percentile=0.5 is same as median when weights equal
     rng = np.random.RandomState(0)
     # Odd size as _weighted_percentile takes lower weighted percentile
     x = rng.randint(10, size=11)
-    weights = np.ones(x.shape)
+    x = x.astype(dtype_name, copy=False)
+    x = xp.asarray(x, device=device)
+    weights = xp.ones(x.shape)
 
     median = np.median(x)
```

Comment:
`xp.median(x)` does not exist in the Array API spec, so I think, since we pass `x` into it, we either need to write our custom function for it or we can pick not to parametrise this test with Array API things, to keep this PR a bit efficient. This also applies for `test_weighted_median_integer_weights` below, where `np.median()` is also used.